src/UXClient/Components/GeoProcessGraphic/GeoProcessGraphic.ts

```diff
@@ -0,0 +1,170 @@
+import { TsqExpression } from '../../Models/TsqExpression';
+import { ServerClient } from '../../../ServerClient/ServerClient';
+
+class GeoProcessGraphic extends HistoryPlayback {
+    private dataSource: atlas.source.DataSource;
+    private marker: atlas.HtmlMarker;
+    private subscriptionKey: string;
+    private img: string;
+    private zoom: number;
+    private width: number;
+    private height: number;
+    private popup: atlas.Popup;
+    private positionXVariableName: string;
+    private positionYVariableName: string;
+    private theme: string;
+    private alias: string;
+
+  constructor(renderTarget: Element){
+    super(renderTarget);
+    this.serverClient = new ServerClient();
+    this.currentCancelTrigger = null;
+  }
+
+  render(environmentFqdn: string,
+    getToken: () => Promise<string>,
+    graphicSrc: string,
+    data: Array<TsqExpression>,
+    chartOptions
+    ) {
+    this.zoom = chartOptions.zoom;
+    this.subscriptionKey = chartOptions.subscriptionKey;
+    this.width = chartOptions.width;
+    this.height = chartOptions.height;
+    this.img = graphicSrc;
+    this.theme= chartOptions.theme;
+    this.renderBase(environmentFqdn, getToken, graphicSrc, data, chartOptions);
+  }
+
+  protected loadGraphic(graphicSrc: string): Promise<GraphicInfo> {
+    (<HTMLElement>this.component.node()).style.width = `${this.width}px`;
+    (<HTMLElement>this.component.node()).style.height = `${this.height}px`;
+    
+    let map = new atlas.Map(<HTMLElement>this.component.node(), {
+      authOptions: {
+        authType: atlas.AuthenticationType.subscriptionKey,
+        subscriptionKey: this.subscriptionKey
+      }
+    });
+    map.events.add('ready', () => {
+        this.dataSource = new atlas.source.DataSource();
+        map.sources.add(this.dataSource);
+        this.popup = new atlas.Popup({
+            content: `<div style="padding:10px;"></div>`,
+            pixelOffset: [0, -30]
+        })
+        this.marker =  new atlas.HtmlMarker({
+            htmlContent: '<img class="circleImage " src= "' + this.img + '" />',
+            position: [0, 0],
+            popup: this.popup
+        });
+        map.markers.add(this.marker);
+        map.events.add('click', this.marker, () => {
+            this.marker.togglePopup();
+        });
+    });
+    map.setCamera({
+        center: [0, 0],
+        bearing: 0,
+        pitch: 90,
+        zoom: this.zoom,
+        maxZoom : 4,
+        minZoom: 1.5,
+        type: "fly",
+        duration: 100
+    });
+    return Promise.resolve({
+        graphic: map,
+        width: this.width,
+        height: this.height
+      });
+  }
+  
+  protected createDataPoints(prm: Array<IGeoProcessGraphicLabelInfo>){
+    let dataPoints = prm.map((r, i): IGeoProcessGraphicLabelInfo => {
+        let result = this.parseTsqResponse(r);
+        this.positionXVariableName = this.tsqExpressions[i].positionXVariableName;
```

Comment:
waittt, doesnt "this" here refer to the process graphic, not to the tsqExpression?  this means that every tsqExpression will have a positionXVariableName as whatever the last tsqExpression is?